modules/nf-core/gsea/gsea/main.nf

```diff
@@ -0,0 +1,77 @@
+    tuple val(meta), path("*/[!gene_set_size|gsea_report|ranked_gene_list]*.tsv")                                                           , emit: gene_set_tsv, optional: true
+    tuple val(meta), path("*/[!gsea_report|heat_map_corr_plot|index|pos_snapshot|neg_snapshot]*.html")                                      , emit: gene_set_html, optional: true
+    tuple val(meta), path("*/[!butterfly|enplot|global_es_histogram|gset_rnd_es_dist|heat_map|pvalues_vs_nes_plot|ranked_list_corr]*.png")  , emit: gene_set_heatmap, optional: true
+    tuple val(meta), path("*/*_snapshot*.html")                                                                                             , emit: snapshot, optional: true
+    tuple val(meta), path("*/enplot*.png")                                                                                                  , emit: gene_set_enplot, optional: true
+    tuple val(meta), path("*/gset_rnd_es_dist*.png")                                                                                        , emit: gene_set_dist, optional: true
+    tuple val(meta), path("*/*.zip")                                                                                                        , emit: archive, optional: true
+    path "versions.yml"                                                                                                                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: [ ]
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    # Run GSEA
+    gsea-cli GSEA \
+        -res $gct \
+        -cls ${cls}#${args2.target}_versus_${args2.reference} \
+        -gmx $gene_sets \
+        -out \$(pwd) \
+        --rpt_label $prefix \
+        $args
+
+    # Move things out of the timestamped folder
+    mv ${prefix}.Gsea.* ${prefix}
+
+    # Other files have the annoying timestamp prefix too, but may be referred
+    # to from other files (e.g. the index.html). Symlinked here to provide
+    # consistent paths for the nf-core CI
+
+    pushd $prefix
+    ln -s ${prefix}.Gsea.*.rpt ${prefix}.Gsea.rpt
+    (ls *.zip 2>/dev/null || true) | while read -r l; do
+        ln -s \$l ${prefix}.Gsea.rpt.zip
+    done
+
+    ln -s \$(ls ranked_gene_list_${args2.target}_versus_${args2.reference}_*.tsv) ranked_gene_list_${args2.target}_versus_${args2.reference}.tsv
+    ln -s \$(ls gsea_report_for_${args2.reference}_*.html) gsea_report_for_reference_${args2.reference}.html
+    ln -s \$(ls gsea_report_for_${args2.reference}_*.tsv) gsea_report_for_reference_${args2.reference}.tsv
+    ln -s \$(ls gsea_report_for_${args2.target}_*.html) gsea_report_for_target_${args2.target}.html
+    ln -s \$(ls gsea_report_for_${args2.target}_*.tsv) gsea_report_for_target_${args2.target}.tsv
+    popd
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        gsea: 4.3.2
```

Comment:
Don't embed directly into versions.yml! Please follow the convention/examples on what to do when a tool doesn't produce them itself [^1]

[^1]: https://nf-co.re/docs/contributing/modules#general point 4, homoe/annotatepeaks example

Reply:
Ahh- thanks, good to know.